Adding resource definition for IoTSecuritySolutionsAnalytics

hagba opened the pull request:
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

hagba committed:
Adding resource definition for IoTSecuritySolutionsAnalytics

AutorestCI commented:
# Automation for azure-sdk-for-ruby

AutorestCI commented:
# Automation for azure-sdk-for-python

AutorestCI commented:
# Automation for azure-sdk-for-java

AutorestCI commented:
# Automation for azure-sdk-for-js

azuresdkci commented:
Can one of the admins verify this patch?

AutorestCI commented:
# Automation for azure-sdk-for-go

nschonni commented on specification/security/resource-manager/Microsoft.Security/preview/2017-08-01-preview/iotSecuritySolutionAnalytics.json:
```diff
@@ -0,0 +1,589 @@
+					"items": {
+
+								"type": "#/definitions/IoTSecurityDeviceAlerts",
+								"description": "top 3 devices with most atacked alerts."
```
```suggestion
								"description": "top 3 devices with most attacked alerts."
```

nschonni commented on specification/security/resource-manager/Microsoft.Security/preview/2017-08-01-preview/iotSecuritySolutionAnalytics.json:
```diff
@@ -0,0 +1,589 @@
+				"count": {
+					"readOnly": true,
+					"type": "int",
+					"description": "Occurance number of the alert within the aggregated date"
```
```suggestion
					"description": "Occurrence number of the alert within the aggregated date"
```

Commit:
Fix reference error

hagba committed:
Add reference to iotSecuritySolutionsAnalytics.json

hagba committed:
Fix Autorest and swagger errors

hagba committed:
Remove paging from short responses

hagba committed:
Fix spelling

hagba committed:
Fix wording of description

hagba committed:
Fix paging error

hagba committed:
refix paging

hagba committed:
add Lists support

hagba committed:
Fix example syntax

hagba committed:
fix schema warnings

hagba committed:
fix schema

hagba committed:
fix schema

hagba committed:
fix alert required properties error

hagba committed:
add properties level for assitional properties

hagba committed:
Fix additional proprties issues

hagba committed:
fix additional properties issues

hagba committed:
remove required tag

majastrz commented on specification/security/resource-manager/Microsoft.Security/preview/2017-08-01-preview/iotSecuritySolutionAnalytics.json:
```diff
@@ -0,0 +1,711 @@
+  "info": {
+    "title": "Security Center",
+    "description": "API spec for Microsoft.Security (Azure Security Center) resource provider",
+    "version": "2019-05-01-preview"
```
>2019-05-01-preview [](start = 16, length = 18)

The version doesn't match the directory name. Which one should it be? `2017-08-01-preview` or `2019-05-01-preview`?

hagba replied:
2017-08-01-preview. fixed

majastrz commented on specification/security/resource-manager/Microsoft.Security/preview/2017-08-01-preview/iotSecuritySolutionAnalytics.json:
```diff
@@ -0,0 +1,711 @@
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Security/iotSecuritySolutions/{solutionName}/Analytics/current": {
```
>current [](start = 147, length = 7)

For a singleton resource, the ARM naming convention is to call it `default`. Can you change the name?

majastrz replied:
The resource type naming convention is camelCasePlural. Can you change the name to `analytics`?

---
In reply to: [295953066](https://github.com/Azure/azure-rest-api-specs/pull/6160#discussion_r295953066) [](ancestors = 295953066)

majastrz replied:
On 2nd thought analytics sounds really vague. I think analyticsModels or analyticModels would fit better and matches the name of the definition.

---
In reply to: [295954442](https://github.com/Azure/azure-rest-api-specs/pull/6160#discussion_r295954442) [](ancestors = 295954442,295953066)

hagba replied:
I understood that when I have only one model data I need to use current.
I have changed it to default

majastrz commented on specification/security/resource-manager/Microsoft.Security/preview/2017-08-01-preview/iotSecuritySolutionAnalytics.json:
```diff
@@ -0,0 +1,711 @@
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Security/iotSecuritySolutions/{solutionName}/Analytics/current/AggregatedAlert": {
```
>AggregatedAlert [](start = 155, length = 15)

Can you rename to `aggregatedAlerts` to match the typical convention?

majastrz commented on specification/security/resource-manager/Microsoft.Security/preview/2017-08-01-preview/iotSecuritySolutionAnalytics.json:
```diff
@@ -0,0 +1,711 @@
+      }
+    }
+  },
+  "paths": {
```
>paths [](start = 3, length = 5)

You're missing the path to get all `analytics` resources. That needs to be implemented even if you just have 1. There are services in azure (Azure Policy is one) that scan and discover resources and they rely on these routes being implemented.

majastrz commented on specification/security/resource-manager/Microsoft.Security/preview/2017-08-01-preview/iotSecuritySolutionAnalytics.json:
```diff
@@ -0,0 +1,711 @@
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Security/iotSecuritySolutions/{solutionName}/Analytics/current/AggregatedAlert/{aggregatedAlertName}/Dismiss": {
```
>Dismiss [](start = 193, length = 7)

Should be lower case `dismiss`

majastrz commented on specification/security/resource-manager/Microsoft.Security/preview/2017-08-01-preview/iotSecuritySolutionAnalytics.json:
```diff
@@ -0,0 +1,711 @@
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Security/iotSecuritySolutions/{solutionName}/Analytics/current/AggregatedAlert/{aggregatedAlertName}/Dismiss": {
+      "put": {
```
>put [](start = 7, length = 3)

Actions on a resource are required to be POST as per the RPC. [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/proxy-api-reference.md#resource-action-requests](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/proxy-api-reference.md#resource-action-requests). PUT is reserved for resource creation and update.

majastrz commented on specification/security/resource-manager/Microsoft.Security/preview/2017-08-01-preview/iotSecuritySolutionAnalytics.json:
```diff
@@ -0,0 +1,711 @@
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Security/iotSecuritySolutions/{solutionName}/Analytics/current/AggregatedRecommendation/{aggregatedRecommendationName}": {
```
>AggregatedRecommendation [](start = 155, length = 24)

Can you rename to `aggregatedRecommendations`?

majastrz commented on specification/security/resource-manager/Microsoft.Security/preview/2017-08-01-preview/iotSecuritySolutionAnalytics.json:
```diff
@@ -0,0 +1,711 @@
+        "value"
+      ],
+      "properties": {
+        "value": {
```
>value [](start = 9, length = 5)

nit - you can make the value array read only also.

majastrz commented on specification/security/resource-manager/Microsoft.Security/preview/2017-08-01-preview/iotSecuritySolutionAnalytics.json:
```diff
@@ -0,0 +1,711 @@
+        }
+      }
+    },
+    "IoTSecuritySolutionAnalyticsModel": {
```
>IoTSecuritySolutionAnalyticsModel [](start = 5, length = 33)

This doesn't look like an azure resource at all. The JSON should include properties like `id`, `name`, `type`, and `properties`. You should make this similar to `IoTSecurityAggregatedAlert` in the sense that you derive from a common resource definition and customize the contents of the `properties` property.

majastrz reviewed (changes requested):
I added comments on the Swagger. Can you address the issues?

hagba commented:
Thanks,
I will start working on them

From: Marcin Jastrzebski <notifications@github.com>
Sent: Thursday, June 20, 2019 10:27 PM
To: Azure/azure-rest-api-specs <azure-rest-api-specs@noreply.github.com>
Cc: Hagit Badash <hagb@microsoft.com>; Author <author@noreply.github.com>
Subject: Re: [Azure/azure-rest-api-specs] [Don't merge] Adding resource definition for IoTSecuritySolutionsAnalytics (#6160)


@majastrz requested changes on this pull request.

I added comments on the Swagger. Can you address the issues?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-rest-api-specs%2Fpull%2F6160%3Femail_source%3Dnotifications%26email_token%3DALTVOKGWKOSRGK5UY7G3CMLP3PKYDA5CNFSM4HRH7RHKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOB4GKUDI%23pullrequestreview-252488205&data=02%7C01%7Chagb%40microsoft.com%7Ca1a6285698bd497b72d208d6f5b542d1%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636966556194793042&sdata=LWfrYwY%2FMMnaNzB5GbZzEISgxBUC3OAuxRxI3Lg%2FGjs%3D&reserved=0>, or mute the thread<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FALTVOKER455D2ECZEH2X4R3P3PKYDANCNFSM4HRH7RHA&data=02%7C01%7Chagb%40microsoft.com%7Ca1a6285698bd497b72d208d6f5b542d1%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636966556194793042&sdata=rxeKJoyrxjDJNytb%2FrEYYR%2Bwrp63uzwlX0DVDQZ7LBE%3D&reserved=0>.

hagba committed:
Fix according to ARM comments

hagba committed:
fix errors

hagba committed:
fix recommendation type example

hagba committed:
fix dismiss example

hagba committed:
required property cannot be readonly

hagba committed:
remove tags from examples

hagba committed:
fix lists errors

hagba committed:
remove location property

hagba commented:
I have changed the files according to the comments and improved the example data.
I get errors in the automation tool that are not related to my changed files.

Thanks,
Hagit.

From: Hagit Badash
Sent: Thursday, June 20, 2019 11:24 PM
To: Azure/azure-rest-api-specs <reply@reply.github.com>; Azure/azure-rest-api-specs <azure-rest-api-specs@noreply.github.com>
Cc: Author <author@noreply.github.com>
Subject: RE: [Azure/azure-rest-api-specs] [Don't merge] Adding resource definition for IoTSecuritySolutionsAnalytics (#6160)

Thanks,
I will start working on them

From: Marcin Jastrzebski <notifications@github.com<mailto:notifications@github.com>>
Sent: Thursday, June 20, 2019 10:27 PM
To: Azure/azure-rest-api-specs <azure-rest-api-specs@noreply.github.com<mailto:azure-rest-api-specs@noreply.github.com>>
Cc: Hagit Badash <hagb@microsoft.com<mailto:hagb@microsoft.com>>; Author <author@noreply.github.com<mailto:author@noreply.github.com>>
Subject: Re: [Azure/azure-rest-api-specs] [Don't merge] Adding resource definition for IoTSecuritySolutionsAnalytics (#6160)


@majastrz requested changes on this pull request.

I added comments on the Swagger. Can you address the issues?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-rest-api-specs%2Fpull%2F6160%3Femail_source%3Dnotifications%26email_token%3DALTVOKGWKOSRGK5UY7G3CMLP3PKYDA5CNFSM4HRH7RHKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOB4GKUDI%23pullrequestreview-252488205&data=02%7C01%7Chagb%40microsoft.com%7Ca1a6285698bd497b72d208d6f5b542d1%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636966556194793042&sdata=LWfrYwY%2FMMnaNzB5GbZzEISgxBUC3OAuxRxI3Lg%2FGjs%3D&reserved=0>, or mute the thread<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FALTVOKER455D2ECZEH2X4R3P3PKYDANCNFSM4HRH7RHA&data=02%7C01%7Chagb%40microsoft.com%7Ca1a6285698bd497b72d208d6f5b542d1%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636966556194793042&sdata=rxeKJoyrxjDJNytb%2FrEYYR%2Bwrp63uzwlX0DVDQZ7LBE%3D&reserved=0>.

Commit:
Merge branch 'master' into dev-security-Microsoft.Security-2017-08-01-preview

Commit:
Fixed securitySolution patch swagger

ravbhatnagar reviewed (commented):
some minor notes. please take a look.

hagba commented on specification/security/resource-manager/Microsoft.Security/preview/2017-08-01-preview/iotSecuritySolutionAnalytics.json:
```diff
@@ -0,0 +1,757 @@
+      "type": "object",
+      "description": "Security Solution Aggregated Alert data",
+      "properties": {
+        "alertType": {
```
I do not want to limit it and update it on every alert that we will support in the future

hagba commented on specification/security/resource-manager/Microsoft.Security/preview/2017-08-01-preview/iotSecuritySolutionAnalytics.json:
```diff
@@ -0,0 +1,757 @@
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/IoTSecuritySolutionAnalyticsModel"
```
I want to emphasize that the list will always contain one value.

ravbhatnagar replied:
yes, thats fine. but in a collection response, ARM expects a value array. You can have an array with one value.

hagba commented on specification/security/resource-manager/Microsoft.Security/preview/2017-08-01-preview/iotSecuritySolutionAnalytics.json:
```diff
@@ -0,0 +1,757 @@
+          "type": "string",
+          "description": "Azure resource ID of the resource that got the alerts"
+        },
+        "systemSource": {
```
It is not enum in the alert object in the alerts model (alerts.json)

hagba committed:
fix according to ARMs comment

hagba committed:
Merge branch 'dev-security-Microsoft.Security-2017-08-01-preview' of https://github.com/hagba/azure-rest-api-specs into dev-security-Microsoft.Security-2017-08-01-preview

hagba committed:
fix unhealthyDeviceCount property name

hagba committed:
Fix syntax error

nschonni commented on specification/security/resource-manager/Microsoft.Security/preview/2017-08-01-preview/iotSecuritySolutionAnalytics.json:
```diff
@@ -0,0 +1,777 @@
+        "recommendationDisplayName": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Display name of the resommendation type"
```
```suggestion
          "description": "Display name of the recommendation type."
```

hagba committed:
Update specification/security/resource-manager/Microsoft.Security/preview/2017-08-01-preview/iotSecuritySolutionAnalytics.json

Co-Authored-By: Nick Schonning <nschonni@gmail.com>

ravbhatnagar commented:
Looks good. Signing off from ARM side.

jhendrixMSFT commented:
@AutorestCI regenerate azure-sdk-for-go

AutorestCI commented:
<details><summary>Encountered a Subprocess error</summary><p>

Command: ['/usr/local/bin/autorest', '/tmp/tmp0tztcfnn/rest/specification/security/resource-manager/readme.md', '--go', '--go-sdk-folder=/tmp/tmp0tztcfnn/src/github.com/Azure/azure-sdk-for-go', '--multiapi', '--preview-chk', '--use=@microsoft.azure/autorest.go@~2.1.133', '--use-onever', '--verbose']
Finished with return code 1
and output:
```shell
AutoRest code generation utility [version: 2.0.4283; node: v8.12.0]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/root/.autorest/@microsoft.azure_autorest-core@2.0.4302/node_modules/@microsoft.azure/autorest-core/dist' (2.0.4302)
   Including configuration file 'file:///tmp/tmp0tztcfnn/rest/specification/security/resource-manager/readme.go.md'
   Loading AutoRest extension '@microsoft.azure/autorest.go' (~2.1.133->2.1.133)
   Loading AutoRest extension '@microsoft.azure/autorest.modeler' (2.3.38->2.3.38)
Processing batch task - {"tag":"package-composite-v1"} .
   Including configuration file 'file:///tmp/tmp0tztcfnn/rest/specification/security/resource-manager/readme.go.md'
Processing batch task - {"tag":"package-composite-v2"} .
   Including configuration file 'file:///tmp/tmp0tztcfnn/rest/specification/security/resource-manager/readme.go.md'
Processing batch task - {"tag":"package-composite-v3"} .
   Including configuration file 'file:///tmp/tmp0tztcfnn/rest/specification/security/resource-manager/readme.go.md'
FATAL: System.InvalidOperationException: Values specified by 'enum' mismatch those specified by 'x-ms-enum' (name: 'RecommendationType'): 'OpenPortsOnDevice', 'PermissiveFirewallPolicy ', 'PermissiveFirewallRuleInput', 'PermissiveFirewallRuleOut', 'OperationSystemNotValid', 'UnutilizedMessagesFromAgent', 'SecurityTwinConfigurationNotOptimal', 'SecurityTwinConfigurationConflict', 'IdenticalAuthenticationCredentials', 'DenyDefaultIpPolicy', 'TooLargeIPRange', 'EnableDiagnosticsLog ' vs 'OpenPortsOnDevice', 'PermissiveFirewallPolicy', 'PermissiveFirewallRuleInput', 'PermissiveFirewallRuleOut', 'OperationSystemNotValid', 'UnutilizedMessagesFromAgent', 'SecurityTwinConfigurationNotOptimal', 'SecurityTwinConfigurationConflict', 'IdenticalAuthenticationCredentials', 'DenyDefaultIpPolicy', 'TooLargeIPRange', 'EnableDiagnosticsLog'
   at AutoRest.Modeler.ObjectBuilder.BuildServiceType(String serviceTypeName, Boolean required) in C:\Users\ci\AppData\Local\Temp\PUBLISH8kiit\38_20171116T010915\autorest.modeler\src\ObjectBuilder.cs:line 129
   at AutoRest.Modeler.SchemaBuilder.ParentBuildServiceType(String serviceTypeName, Boolean required) in C:\Users\ci\AppData\Local\Temp\PUBLISH8kiit\38_20171116T010915\autorest.modeler\src\SchemaBuilder.cs:line 204
   at AutoRest.Modeler.SchemaBuilder.BuildServiceType(String serviceTypeName, Boolean required) in C:\Users\ci\AppData\Local\Temp\PUBLISH8kiit\38_20171116T010915\autorest.modeler\src\SchemaBuilder.cs:line 46
   at AutoRest.Modeler.SchemaBuilder.BuildServiceType(String serviceTypeName, Boolean required) in C:\Users\ci\AppData\Local\Temp\PUBLISH8kiit\38_20171116T010915\autorest.modeler\src\SchemaBuilder.cs:line 131
   at AutoRest.Modeler.ObjectBuilder.BuildServiceType(String serviceTypeName, Boolean required) in C:\Users\ci\AppData\Local\Temp\PUBLISH8kiit\38_20171116T010915\autorest.modeler\src\ObjectBuilder.cs:line 179
   at AutoRest.Modeler.SchemaBuilder.ParentBuildServiceType(String serviceTypeName, Boolean required) in C:\Users\ci\AppData\Local\Temp\PUBLISH8kiit\38_20171116T010915\autorest.modeler\src\SchemaBuilder.cs:line 204
   at AutoRest.Modeler.SchemaBuilder.BuildServiceType(String serviceTypeName, Boolean required) in C:\Users\ci\AppData\Local\Temp\PUBLISH8kiit\38_20171116T010915\autorest.modeler\src\SchemaBuilder.cs:line 46
   at AutoRest.Modeler.SchemaBuilder.BuildServiceType(String serviceTypeName, Boolean required) in C:\Users\ci\AppData\Local\Temp\PUBLISH8kiit\38_20171116T010915\autorest.modeler\src\SchemaBuilder.cs:line 131
   at AutoRest.Modeler.SchemaBuilder.BuildServiceType(String serviceTypeName, Boolean required) in C:\Users\ci\AppData\Local\Temp\PUBLISH8kiit\38_20171116T010915\autorest.modeler\src\SchemaBuilder.cs:line 131
   at AutoRest.Modeler.ObjectBuilder.BuildServiceType(String serviceTypeName, Boolean required) in C:\Users\ci\AppData\Local\Temp\PUBLISH8kiit\38_20171116T010915\autorest.modeler\src\ObjectBuilder.cs:line 179
   at AutoRest.Modeler.SchemaBuilder.ParentBuildServiceType(String serviceTypeName, Boolean required) in C:\Users\ci\AppData\Local\Temp\PUBLISH8kiit\38_20171116T010915\autorest.modeler\src\SchemaBuilder.cs:line 204
   at AutoRest.Modeler.SchemaBuilder.BuildServiceType(String serviceTypeName, Boolean required) in C:\Users\ci\AppData\Local\Temp\PUBLISH8kiit\38_20171116T010915\autorest.modeler\src\SchemaBuilder.cs:line 46
   at AutoRest.Modeler.SchemaBuilder.BuildServiceType(String serviceTypeName, Boolean required) in C:\Users\ci\AppData\Local\Temp\PUBLISH8kiit\38_20171116T010915\autorest.modeler\src\SchemaBuilder.cs:line 131
   at AutoRest.Modeler.SwaggerModeler.BuildCompositeTypes() in C:\Users\ci\AppData\Local\Temp\PUBLISH8kiit\38_20171116T010915\autorest.modeler\src\SwaggerModeler.cs:line 271
   at AutoRest.Modeler.SwaggerModeler.Build(ServiceDefinition serviceDefinition) in C:\Users\ci\AppData\Local\Temp\PUBLISH8kiit\38_20171116T010915\autorest.modeler\src\SwaggerModeler.cs:line 66
   at AutoRest.Modeler.Program.<ProcessInternal>d__2.MoveNext() in C:\Users\ci\AppData\Local\Temp\PUBLISH8kiit\38_20171116T010915\autorest.modeler\src\Program.cs:line 60
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NewPlugin.<Process>d__15.MoveNext()
FATAL: go/imodeler1 - FAILED
FATAL: Error: Plugin imodeler1 reported failure.
Process() cancelled due to exception : Plugin imodeler1 reported failure.
Failure during batch task - {"tag":"package-composite-v3"} -- Error: Plugin imodeler1 reported failure..
  Error: Plugin imodeler1 reported failure.
```

</p></details>

hagba committed:
Move remediation steps from alerts to recommendations

hagba committed:
Move remediation steps from alerts to recommendations

hagba committed:
Merge branch 'dev-security-Microsoft.Security-2017-08-01-preview' of https://github.com/hagba/azure-rest-api-specs into dev-security-Microsoft.Security-2017-08-01-preview

nschonni commented on package.json:
```diff
@@ -50,5 +50,8 @@
     "tsc": "tsc",
     "multiapi": "ts-node ./scripts/multiapi.ts"
+  },
+  "dependencies": {
```
Is this needed?

hagba committed:
rename mostPrevalentDevices to mostPrevalentDeviceAlerts

hagba committed:
rename mostPrevalentDevices to mostPrevalentDeviceAlerts

hagba committed:
add mostPrevalentDeviceRecommendations

hagba committed:
fix example

hagba committed:
fix dismiss

hagba committed:
add severity to prevalent lists

hagba committed:
fix syntax

hagba committed:
fix errors

hagba committed:
fix enum

sergey-shandar commented on package.json:
```diff
@@ -50,5 +50,8 @@
     "multiapi": "ts-node ./scripts/multiapi.ts"
+  },
+  "dependencies": {
+    "autorest": "^2.0.4283"
```
please, remove this line

hagba committed:
remove dependency

sergey-shandar commented:
@majastrz please, review the PR.

majastrz reviewed (approved):
Approving since Gaurav already marked it as ARMSignedOff.
